boilerplate/app/navigators/navigationUtilities.ts

```diff
@@ -103,16 +93,17 @@ function navigationRestoredDefaultState(persistNavigation: PersistNavigationConf
 /**
  * Custom hook for persisting navigation state.
  */
-export function useNavigationPersistence(storage: any, persistenceKey: string) {
-  const [initialNavigationState, setInitialNavigationState] = useState()
+export function useNavigationPersistence(storage: Storage, persistenceKey: string) {
+  const [initialNavigationState, setInitialNavigationState] =
+    useState<NavigationProps["initialState"]>()
   const isMounted = useIsMounted()
 
   const initNavState = navigationRestoredDefaultState(Config.persistNavigation)
   const [isRestored, setIsRestored] = useState(initNavState)
 
-  const routeNameRef = useRef<string | undefined>()
+  const routeNameRef = useRef<keyof AppStackParamList | undefined>()
```

Comment:
If there were other navigators involved would you union those param lists here as well? For example the ones from the Demo Navigator in the case of the demo boilerplate

Reply:
Yeah we'd want to include that here, since we are referencing the root navigator being passed around.

Reply:
Pushed up a commit to add some comments to this: https://github.com/infinitered/ignite/pull/2424/commits/dbce8f4d91152ca845afcd7c6060342541cf2749

Reply:
@frankcalise you good with the comment on this? Then I'll be good to merge